includes/class-amp-post-type-support.php

```diff
@@ -123,7 +123,12 @@ public static function get_support_errors( $post ) {
 			$post = get_post( $post );
 		}
 		$errors = [];
-
+		
+		if ( ! isset( $post->post_type ) && ! isset( $post->ID ) ) {
```

Comment:
This could be simplified to:

```suggestion
		if ( ! $post ) {
```

Reply:
Or what about:

```suggestion
		if ( ! $post instanceof WP_Post ) {
```

Reply:
Yes that's more appropriate.